testing/test-setup-config/src/lib/vitest-config-factory.unit.test.ts

```diff
@@ -0,0 +1,356 @@
+            include: [
+              'src/**/*.unit.test.{js,mjs,cjs,ts,mts,cts,jsx,tsx}',
+              'src/**/*.type.test.{js,mjs,cjs,ts,mts,cts,jsx,tsx}',
+            ],
+            globalSetup: ['../../global-setup.ts'],
+            setupFiles: expect.arrayContaining([
+              '../../testing/test-setup/src/lib/console.mock.ts',
+              '../../testing/test-setup/src/lib/reset.mocks.ts',
+              '../../testing/test-setup/src/lib/fs.mock.ts',
+            ]),
+            coverage: expect.objectContaining({
+              reporter: ['text', 'lcov'],
+              reportsDirectory: '../../coverage/test-package/unit-tests',
+              exclude: ['mocks/**', '**/types.ts', 'perf/**'],
+            }),
+            typecheck: { include: ['**/*.type.test.ts'] },
+          }),
+        }),
+      );
+      expect(defineConfig).toHaveBeenCalledWith(config);
+    });
+
+    it('should include all required setup files for unit tests', () => {
+      const config = createVitestConfig('test-package', 'unit');
+
+      const setupFiles = (config as any).test.setupFiles;
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/console.mock.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/reset.mocks.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/cliui.mock.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/fs.mock.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/git.mock.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/portal-client.mock.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/extend/ui-logger.matcher.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/extend/markdown-table.matcher.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/extend/jest-extended.matcher.ts',
+      );
+      expect(setupFiles).toContain(
+        '../../testing/test-setup/src/lib/extend/path.matcher.ts',
+      );
+    });
+
+    it('should include type test pattern in unit tests', () => {
+      const config = createVitestConfig('test-package', 'unit');
+
+      expect((config as any).test.include).toContain(
+        'src/**/*.type.test.{js,mjs,cjs,ts,mts,cts,jsx,tsx}',
+      );
+    });
+
+    it('should enable typecheck for unit tests', () => {
+      const config = createVitestConfig('test-package', 'unit');
+
+      expect((config as any).test.typecheck).toEqual({
+        include: ['**/*.type.test.ts'],
+      });
+    });
+
+    it('should always include perf/** in coverage exclusions', () => {
+      const config = createVitestConfig('test-package', 'unit');
+
+      expect((config as any).test.coverage.exclude).toContain('perf/**');
```

Comment:
There are a lot of unnecessary `config as any` castings in this test suite. To access properties under `config.test`, use `!` [^1] or `?.`.

[^1]: Non-null assertion with `!` isn't safe to use in general, but in tests it doesn't really matter - a runtime error is much the same as a failing test.